### PR TITLE
Complete: Task 609-616: DailyDialog and generated_reviews_enth Dataset

### PR DESCRIPTION
**Pull request by karannaik3797**

### CODE REVIEW

**Review by swarooprm — COMMENTED** [review]
Here are the comments. 

1.  Don't change test_all.py and also don't delete existing files (task1001 and 1002) both from the task folder and readme.
2. Note that, negative examples mean incorrect annotation. E.g. task609, both good and bad reviews should be part of both positive and negative examples. This problem exists in all 4 tasks.
3. task609, 610: Definition is incorrect because I don't see any information about ratings (1-5) in the reviews, also there are redundant things such as CTRL etc.
4. task611-612: Definition needs rephrasing, currently its a bit verbose and complex to read. Show it to your friend and find out if they can understand. 
5. Labels ('Yes' and 'No') across instances should be balanced.

**Comment by swarooprm** [comment]
@karannaik3797, let me know once you incorporate these comments and I will review it again.

**Comment by karannaik3797** [comment]
> @karannaik3797, let me know once you incorporate these comments and I will review it again.

Hey when I do not make that changes in the test_all.py of adding encoding=UTF-8
 then there is this error occurring 
Screenshot (253)
Can you tell me how can I resolve it without changing the test_all.py

**Comment by swarooprm** [comment]
I think the error is because of some issue in creating the json file.
I am not sure how you are creating json, can you use the following command to create json:
json.dumps(your_json_string, indent=4, ensure_ascii=False))

**Comment by karannaik3797** [comment]
> I think the error is because of some issue in creating the json file.
> I am not sure how you are creating json, can you use the following command to create json:
> json.dumps(your_json_string, indent=4, ensure_ascii=False))

I am using the same json.dumps(your_json_string, indent=4, ensure_ascii=False)) function.
Screenshot (254)

**Comment by swarooprm** [comment]
Can you remove (..,encoding="utf8") in line 2 of the screenshot you have sent me? (with open..) and try testing again?

**Comment by karannaik3797** [comment]
> Can you remove (..,encoding="utf8") in line 2 of the screenshot you have sent me? (with open..) and try testing again?

I have removed the encoding = "utf8" and now three files are working without making changes to test_all.py file but one file which has the Thai language is not working and giving an error: UnicodeDecodeError: 'charmap' codec can't decode byte 0x8d in position 1382: character maps to <undefined>

**Comment by swarooprm** [comment]
@Mihir3009, @karannaik3797 is facing an error in parsing thai language text, 
I thought I will ask you, since you have similar experience in parsing languages other than English? Have you seen this error?

**Comment by Mihir3009** [comment]
I never came across this error. 

@karannaik3797 is ans[0] a final dictionary that you want to save?

**Comment by karannaik3797** [comment]
Yes its the final dictionary that has all the json data.

**Comment by swarooprm** [comment]
Check your code when you parse data from the dataset (make sure you are not reading the original data with utf8 encoding) and see if you can debug.
Alternatively, drop that task and create a different task.

**Comment by swarooprm** [comment]
There are several major comments, it would be great if you can come to office hour tomorrow and I can explain. Alternatively, i will comment here if you can't make it to office hour

**Comment by karannaik3797** [comment]
Can you tell me the changes I have to make so that I can do that because I
was not able to attend the office hour.

**Comment by swarooprm** [comment]
@kuntalkumarpal, wondering if you can review this task?

**Comment by kuntalkumarpal** [comment]
> @kuntalkumarpal, wondering if you can review this task?

yes

File: tasks/README.md [hunk 1]
```diff
@@ -6,4 +6,8 @@ This directory contains the tasks that are part of this benchmark.
 Name | Summary | Category
 ---- | ----------- | --------
 `task1001_squad1.1_question_generation` | Generating guestions (based on SQuAD 1.1) | Question Generation  
-`task1002_squad1.1_answer_generation` | Generating answers to SQuAD 1.1 questions | Answer Generation
\ No newline at end of file
+`task1002_squad1.1_answer_generation` | Generating answers to SQuAD 1.1 questions | Answer Generation
+`task609_generated_reviews_enth_classification` | Classifying Good Review or Bad Reviews | Classification  
+`task610_DailyDialog_classification` | Classifying Data of DailyDialog whether the topic of conversation is Tourism or Not in the dialogue | Translation
```
Comment (kuntalkumarpal) [inline tasks/README.md:11]: lowercap name of files

File: tasks/README.md [hunk 2]
```diff
@@ -6,4 +6,8 @@ This directory contains the tasks that are part of this benchmark.
...
+`task611_DailyDialog_classification` | Classifying Data of DailyDialog whether happy or sad emotion in the dialogue | Classification  
+`task612_DailyDialog_classification` | Classifying Data of DailyDialog whether there is question in the dialogue or not | Classification
```
Comment (kuntalkumarpal) [inline tasks/README.md:13]: lowercap name of files

File: tasks/README.md [hunk 3]
```diff
@@ -6,4 +6,8 @@ This directory contains the tasks that are part of this benchmark.
 Name | Summary | Category
 ---- | ----------- | --------
 `task1001_squad1.1_question_generation` | Generating guestions (based on SQuAD 1.1) | Question Generation  
-`task1002_squad1.1_answer_generation` | Generating answers to SQuAD 1.1 questions | Answer Generation
\ No newline at end of file
+`task1002_squad1.1_answer_generation` | Generating answers to SQuAD 1.1 questions | Answer Generation
+`task609_generated_reviews_enth_classification` | Classifying Good Review or Bad Reviews | Classification  
+`task610_DailyDialog_classification` | Classifying Data of DailyDialog whether the topic of conversation is Tourism or Not in the dialogue | Translation
+`task611_DailyDialog_classification` | Classifying Data of DailyDialog whether happy or sad emotion in the dialogue | Classification
```
Comment (kuntalkumarpal) [inline tasks/README.md:12]: lowercap name of files

Comment (kuntalkumarpal) [reply tasks/README.md:12]: add : Classifying Data of DailyDialog whether "there is "

**Comment by kuntalkumarpal** [comment]
Task 609 : keep consistent names "Good Review", "Bad Review". Check the capitalization in the file

**Comment by kuntalkumarpal** [comment]
Task 609 : The definition you have written "In this task, we are given a Dataset having english language product reviews generated from CTRL and then those reviews are translated to Thai languague using Google's Translate Api and annotated as accepted or rejected (correct) based on fluency and adequacy of the translation by human annotators. Here the review of the product is also given review_star. Also, we have used following data preprocessing (i) Drop duplciates on en,th,review_star,correct (ii) duplicates might exist because the translation checking is done by annotators. (iii) Remove reviews that are not between 1-5 stars. (iv) Remove reviews whose correct are not 0 or 1. (v)Deduplicate on en which contains the source sentences. So, in this task, we have translated the englisht language to Thai language for product review and accepted the translation on following criteria (i) If the translated Thai language is accepted by Human annotators based on fluency and adequacy of the translation. Now if the product review_star is greater than equal 3 then it is considered Good reviews and if it is accepted by the human annotators then we are considering it to be positive example  otherwise negative example. Also, if the review_star less than 3 then it is called Bad reviews and if it is accepted by human annotators then it is good example otherwise bad"

Here many things are irrelevant when u see the instances. Also review stars are not available in the instances so the model has to read the review and determine if it is good or bad.

Q : What do you consider as Good and Bad review ? is it same as positive and negative reviews or is it based on good language which are grammatically correct ? This is not clear from the instances

Suggested Definition: Given an English language product review, determine if it is a Good Review or a Bad Review. A good review is one where ........... and Bad review is one where ........

**Comment by kuntalkumarpal** [comment]
In this negative example `{
            "input": "This product is completely useless. It cannot be used with any of the 3 cell batteries listed on the page. The charger itself appears to heat up, so I can see if it could take a charge. Not sure how they would work in another product, but these don't allow you use them in an NP-E4U Battery Grip or any other brand that uses those batteries. Don't waste your money! And make absolutely sure yours won't damage your charger.",
            "output": "Bad review",
            "explanation": "Here the translated sentences is not accepted by the human annotators hence it is considered negative examples having a bad review"
        },`

I do not understand the explanation, input is the review itself, so the explanation does not make sense to me. Write why you think this is a negative example and also why do you think it should be good review

**Comment by kuntalkumarpal** [comment]
Also write the counts of yes and no questions in each of the yes/no datasets.

**Comment by karannaik3797** [comment]
> Also write the counts of yes and no questions in each of the yes/no datasets.

where can write the counts of yes or no in those datasets. Should I make an extra field in the json file naming "yesCount" and "noCount"

**Comment by kuntalkumarpal** [comment]
> > Also write the counts of yes and no questions in each of the yes/no datasets.
> 
> where can write the counts of yes or no in those datasets. Should I make an extra field in the json file naming "yesCount" and "noCount"

Here in comments.

**Comment by karannaik3797** [comment]
> > > Also write the counts of yes and no questions in each of the yes/no datasets.
> > 
> > 
> > where can write the counts of yes or no in those datasets. Should I make an extra field in the json file naming "yesCount" and "noCount"
> 
> Here in comments.

I have made changes as requested by you. Also in the new commits the number of yescount is 100 and the number of nocount is 100 in all the three files.

File: tasks/README.md [hunk 4]
```diff
@@ -6,4 +6,8 @@ This directory contains the tasks that are part of this benchmark.
 Name | Summary | Category
 ---- | ----------- | --------
 `task1001_squad1.1_question_generation` | Generating guestions (based on SQuAD 1.1) | Question Generation  
-`task1002_squad1.1_answer_generation` | Generating answers to SQuAD 1.1 questions | Answer Generation
\ No newline at end of file
+`task1002_squad1.1_answer_generation` | Generating answers to SQuAD 1.1 questions | Answer Generation
+`task609_generated_reviews_enth_classification` | Classifying generated_reviews_enth whether the review is Good Review or Bad Review | Classification
+`task610_dailydialog_classification` | Classifying Data of DailyDialog whether the topic of conversation is Tourism or Not in the dialogue | Classification
+`task611_dailydialog_classification` | Classifying Data of DailyDialog whether happy or sad emotion in the dialogue | Classification
+`task612_dailydialog_classification` | Classifying Data of DailyDialog whether there is question in the dialogue or not | Classification
```
Comment (kuntalkumarpal) [inline tasks/README.md:13]: All 3 file names are same. The tasks are different so give appropriate names

**Comment by kuntalkumarpal** [comment]
@swarooprm looks good to me. If you have any comments

**Comment by karannaik3797** [comment]
> 
> 
> @swarooprm looks good to me except the same file name for 3 tasks. If you have any comments

I have changed the name of all the files as mentioned by kuntal

**Comment by swarooprm** [comment]
Your HW2 is complete.

Would you like to submit the same tasks to our big project repo https://github.com/allenai/natural-instructions-expansion/pulls? You just need to fork that repo, rename tasks starting from the latest number in the repo (which at this point is 591) and create PR there. This way, you can be part of our research on "learning from instructions instead of learning from examples (conventional machine learning)". This is purely optional and your HW2 grade does not depend on this @karannaik3797

**Comment by swarooprm** [comment]
@karannaik3797 Just a reminder that, if you plan to submit these tasks to the main Github repo and join our research project, Monday (27th Sept) is the deadline. Again, this is optional and independent of your HW2 grade.

**Comment by karannaik3797** [comment]
Hey I have created PR in the main Github repo.

**Pull request closed without merging**